### DISCUSSION TITLE
DeepMD-kit(V2.2.2) cannot reshape an array of size 0 into a shape during training

### DISCUSSION THREAD

**Discussion by liqiang4567** [deepmodeling/deepmd-kit#2677] [Q&A]
Hello, when I use DeepMD-kit(V2.2.2), I use the command dp train input.json, and the following error occurs. How can I solve it? Data set I am not using force, is force necessary during training?

error message:
DEEPMD INFO    training data with min nbor dist: 0.9860776715464545
DEEPMD INFO    training data with max nbor size: [2 1]
DEEPMD INFO     _____               _____   __  __  _____           _     _  _   
DEEPMD INFO    |  __ \             |  __ \ |  \/  ||  __ \         | |   (_)| |  
DEEPMD INFO    | |  | |  ___   ___ | |__) || \  / || |  | | ______ | | __ _ | |_ 
DEEPMD INFO    | |  | | / _ \ / _ \|  ___/ | |\/| || |  | ||______|| |/ /| || __|
DEEPMD INFO    | |__| ||  __/|  __/| |     | |  | || |__| |        |   < | || |_ 
DEEPMD INFO    |_____/  \___| \___||_|     |_|  |_||_____/         |_|\_\|_| \__|
DEEPMD INFO    Please read and cite:
DEEPMD INFO    Wang, Zhang, Han and E, Comput.Phys.Comm. 228, 178-184 (2018)
DEEPMD INFO    Zeng et al, arXiv:2304.09409
DEEPMD INFO    See https://deepmd.rtfd.io/credits/ for details.
DEEPMD INFO    installed to:         /home/conda/feedstock_root/build_artifacts/deepmd-kit_1684940057856/work/_skbuild/linux-x86_64-3.10/cmake-install
DEEPMD INFO    source :              v2.2.2
DEEPMD INFO    source brach:         HEAD
DEEPMD INFO    source commit:        92ca097b
DEEPMD INFO    source commit at:     2023-05-24 13:45:03 +0800
DEEPMD INFO    build float prec:     double
DEEPMD INFO    build variant:        cuda
DEEPMD INFO    build with tf inc:    /home/XXX/software/deepmd-kit/lib/python3.10/site-packages/tensorflow/include;/home/XXX/software/deepmd-kit/lib/python3.10/site-packages/tensorflow/../../../../include
DEEPMD INFO    build with tf lib:    
DEEPMD INFO    ---Summary of the training---------------------------------------
DEEPMD INFO    running on:           gpu04
DEEPMD INFO    computing device:     gpu:0
DEEPMD INFO    CUDA_VISIBLE_DEVICES: unset
DEEPMD INFO    Count of visible GPU: 8
DEEPMD INFO    num_intra_threads:    0
DEEPMD INFO    num_inter_threads:    0
DEEPMD INFO    -----------------------------------------------------------------
DEEPMD INFO    ---Summary of DataSystem: training     -----------------------------------------------
DEEPMD INFO    found 1 system(s):
DEEPMD INFO                                        system  natoms  bch_sz   n_bch   prob  pbc
DEEPMD INFO                      ../00.data/training_data       3      11     453  1.000    T
DEEPMD INFO    --------------------------------------------------------------------------------------
DEEPMD INFO    ---Summary of DataSystem: validation   -----------------------------------------------
DEEPMD INFO    found 1 system(s):
DEEPMD INFO                                        system  natoms  bch_sz   n_bch   prob  pbc
DEEPMD INFO                    ../00.data/validation_data       3      11      90  1.000    T
DEEPMD INFO    --------------------------------------------------------------------------------------
DEEPMD INFO    training without frame parameter
DEEPMD INFO    data stating... (this step may take long time)
DEEPMD ERROR   cannot reshape array of size 0 into shape (4986,9)
DEEPMD ERROR   This error may occur when your label mismatch it's name, i.e. you might store global tensor in `atomic_tensor.npy` or atomic tensor in `tensor.npy`.
Traceback (most recent call last):
  File "/home/qli/software/deepmd-kit/lib/python3.10/site-packages/deepmd/utils/data.py", line 559, in _load_data
    data = np.reshape(data, [nframes, ndof])
  File "<__array_function__ internals>", line 180, in reshape
  File "/home/qli/software/deepmd-kit/lib/python3.10/site-packages/numpy/core/fromnumeric.py", line 298, in reshape
    return _wrapfunc(a, 'reshape', newshape, order=order)
  File "/home/qli/software/deepmd-kit/lib/python3.10/site-packages/numpy/core/fromnumeric.py", line 57, in _wrapfunc
    return bound(*args, **kwds)
ValueError: cannot reshape array of size 0 into shape (4986,9)

input file:
{
    "_comment": " model parameters",
    "model": {
	"type_map":	["cl", "h"],
	"descriptor" :{
	    "type":		"se_e2_a",
	    "sel":		"auto",
	    "rcut_smth":	0.50,
	    "rcut":		6.00,
	    "neuron":		[25, 50, 100],
	    "resnet_dt":	false,
	    "axis_neuron":	16,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"fitting_net" : {
	    "neuron":		[240, 240, 240],
	    "resnet_dt":	true,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"_comment":	" that's all"
    },

    "learning_rate" :{
	"type":		"exp",
	"decay_steps":	500,
	"start_lr":	0.001,	
	"stop_lr":	3.51e-8,
	"_comment":	"that's all"
    },

    "loss" :{
	"type":		"ener",
	"start_pref_e":	0.02,
	"limit_pref_e":	1,
	"start_pref_f":	0,
	"limit_pref_f":	0,
	"start_pref_v":	0,
	"limit_pref_v":	0,
	"_comment":	" that's all"
    },

    "training" : {
	"training_data": {
	    "systems":     ["../00.data/training_data"],
	    "batch_size":  "auto",
	    "_comment":	   "that's all"
	},
	"validation_data":{
	    "systems":	   ["../00.data/validation_data"],
	    "batch_size":  "auto",
	    "numb_btch":   1,
	    "_comment":	   "that's all"
	},
	"numb_steps":	100000,
	"seed":		10,
	"disp_file":	"lcurve.out",
	"disp_freq":	200,
	"save_freq":	1000,
	"_comment":	"that's all"
    },    

    "_comment":		"that's all"
}

**Comment by njzjz** [comment]
Please check your data. From the error message, the box seems to be an empty array.

**Comment by liqiang4567** [comment]
Hi, I checked the data, printed the first three frames, and I didn't see anything wrong.
cells:
[[[25.  0.  0.]
  [ 0. 25.  0.]
  [ 0.  0. 25.]]

 [[25.  0.  0.]
  [ 0. 25.  0.]
  [ 0.  0. 25.]]

 [[25.  0.  0.]
  [ 0. 25.  0.]
  [ 0.  0. 25.]]]
coords:
[[[-1.1         0.          0.        ]
  [ 0.7         1.2124356   0.        ]
  [ 0.          0.          0.        ]]

 [[-1.1         0.          0.        ]
  [ 0.43262377  1.3314792   0.        ]
  [ 0.          0.          0.        ]]

 [[-1.2         0.          0.        ]
  [ 1.1         0.          0.        ]
  [ 0.          0.          0.        ]]]
energies:
[3.7458065 4.682527  4.710594 ]
forces:
[]

**Reply by njzjz** [reply]
Obviously, the force is empty.

**Reply by liqiang4567** [reply]
As mentioned earlier we did not use force, is force necessary for dp training?

**Reply by njzjz** [reply]
You don't have to provide the force. If you provide the force, it should not be empty.
